### PR TITLE
Switch to Airbnb's react-dates and add localization

### PR DESCRIPTION
**Pull request by brandones**
I have fixed one style problem by introducing another.

Screenshot from 2020-04-10 18-18-35

Figure we can push this and style based on new feedback from Dave. I figure changing the turquoise to purple will be top of the list. However, I think the multi-month view with a range-picker is pretty nice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,8 +67,10 @@
     "jest": "^24.8.0",
     "lint-staged": "^9.4.3",
     "mockdate": "^2.0.5",
+    "moment": "^2.24.0",
     "prettier": "^1.19.1",
     "pretty-quick": "^1.11.1",
+    "react-dates": "^21.8.0",
```
Comment (brandones) [inline package.json:73]: react-dates relies on moment, which unfortunately means that right now we have both moment and dayjs. We can optimize later.

Comment (mogoodrich) [reply package.json:73]: Funny, moment is generally the standard we had been using within our Javascript within the UI Framework (prior to OWAs or spa).  Then when doing OWA developent, date-fns became the preferred format, so we switched to what.  And then SPA chose dayjs.  And now we have to do back to moment to use React-Dates.

I had been perfectly happy with moment, *but* I believe a drawback of it is that it is all-or-nothing... you need to bundle the whole library in your app, not just the components you need?

**Review by mogoodrich — APPROVED** [review]
LGTM... agree we should merge and then debate the styling... I'm assuming the turquoise won't be too hard a fix?

Also, if you haven't already done so, might make sense to to reach out to the SPA team and see if there's a datepicker we think we should standardize on?

File: package.json (hunk 1 above)
Comment (brandones) [reply package.json:73]: Yeah, date-fns and dayjs seem like the good options. [This](https://github.com/you-dont-need/You-Dont-Need-Momentjs#brief-comparison) is a helpful comparison.

**Comment by brandones** [comment]
Yeah, seems like [overriding styles](https://github.com/airbnb/react-dates#overriding-styles) is well documented.

SPA has a datepicker ([Styleguide](https://styleguide.openmrs.org/?path=/story/openmrs-styleguide--forms)). I'll ask at the next meeting if the inability to specify a display format is a problem for other orgs. But I hope not -- I think locale-specific display is totally reasonable, and the native HTML date input is the future, and should be preferred whenever possible.

**Comment by mogoodrich** [comment]
@brandones oh, right, so to be clear, you mean that SPA just uses the native date-picker?

**Comment by mogoodrich** [comment]
And, yes @brandones, definitely interested in what other users thing...when you bring it up, please highlight that our main concern is the 4/5/2010 vs 5/4/2010, and that with the HTML date tag this would be based on the browser's language setting/locale.

**Comment by brandones** [comment]
Yes, the native HTML datepicker plus custom CSS.

And yes, will do.

**Pull request merged**